Revit_Core_Engine/Query/Bounds.cs

```diff
@@ -45,16 +46,39 @@ public static BoundingBoxXYZ Bounds(this IEnumerable<Solid> solids, Transform tr
             if (solids == null || !solids.Any())
                 return null;
 
-            Solid union = solids.First();
-            foreach (Solid solid in solids.Skip(1))
+            double minX = double.MaxValue;
+            double minY = double.MaxValue;
+            double minZ = double.MaxValue;
+            double maxX = double.MinValue;
+            double maxY = double.MinValue;
+            double maxZ = double.MinValue;
+
+            foreach (Solid solid in solids)
             {
-                union = BooleanOperationsUtils.ExecuteBooleanOperation(union, solid, BooleanOperationsType.Union);
+                BoundingBoxXYZ solidBBox = solid.GetBoundingBox();
+                XYZ solidMin = solidBBox.Min;
+                XYZ solidMax = solidBBox.Max;
+                XYZ solidOrigin = solidBBox.Transform.Origin;
+
+                minX = Math.Min(minX, solidMin.X + solidOrigin.X);
+                minY = Math.Min(minY, solidMin.Y + solidOrigin.Y);
+                minZ = Math.Min(minZ, solidMin.Z + solidOrigin.Z);
+                maxX = Math.Max(maxX, solidMax.X + solidOrigin.X);
+                maxY = Math.Max(maxY, solidMax.Y + solidOrigin.Y);
+                maxZ = Math.Max(maxZ, solidMax.Z + solidOrigin.Z);
             }
 
+            BoundingBoxXYZ unionBbox = new BoundingBoxXYZ();
+            unionBbox.Min = new XYZ(minX, minY, minZ);
+            unionBbox.Max = new XYZ(maxX, maxY, maxZ);
+
             if (transform != null)
-                union = SolidUtils.CreateTransformed(union, transform);
+            {
+                Solid union = SolidUtils.CreateTransformed(unionBbox.ToSolid(), transform);
+                unionBbox = union.GetBoundingBox();
+            }
```

Comment:
I am afraid this may not be correct: in your case, the resultant bounding box will be red (bounding box of a green bounding box), while it should be blue.

![image](https://user-images.githubusercontent.com/26874773/178990058-431f9475-6a4f-4787-b6ee-ce16100d0cb0.png)

Instead, I would rather transform each solid inside the `foreach` loop.

Reply:
Yes, that still need to be corrected. Thanks! 👍